contracts/upgradeable_contracts/ChaiConnector.sol

```diff
@@ -0,0 +1,156 @@
+pragma solidity 0.4.24;
+
+import "../interfaces/IChai.sol";
+import "./Ownable.sol";
+import "./ERC20Bridge.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+/**
+* @title ChaiConnector
+* @dev This logic allows to use Chai token (https://github.com/dapphub/chai)
+*/
+contract ChaiConnector is Ownable, ERC20Bridge {
+    using SafeMath for uint256;
+
+    bytes32 internal constant CHAI_TOKEN = 0xe529dd1fa310362a861f9a51ed0d07b46ef28d89054300cd2734814ddfcfd449; // keccak256(abi.encodePacked("chaiToken"))
+    bytes32 internal constant INVESTED_AMOUNT = 0xb6afb3323c9d7dc0e9dab5d34c3a1d1ae7739d2224c048d4ee7675d3c759dd1b; // keccak256(abi.encodePacked("investedAmount"))
+
+    uint256 internal constant ONE = 10**27;
+
+    /**
+    * @dev Fixed point multiplication
+    * @return Truncated value of x * y
+    */
+    function rmul(uint256 x, uint256 y) internal pure returns (uint256) {
+        return x.mul(y) / ONE;
+    }
+
+    /**
+    * @dev Fixed point division
+    * @return Ceiled value of x / y
+    */
+    function rdivup(uint256 x, uint256 y) internal pure returns (uint256) {
+        return x.mul(ONE).add(y.sub(1)) / y;
+    }
+
+    /**
+    * @dev Initializes chai token
+    * @param _chai Chai token contract address
+    */
+    function initializeChaiToken(address _chai) external onlyOwner {
```

Comment:
Consider to hardcode the address of Chai token contract.

Reply:
Should we still have a functionality for enabling/disabling chai token then?

Reply:
Yes, calling of `initializeChaiToken` will enable the functionality (the Chai token contract address will be hardcoded here instead of specifying be the transaction sender). `removeChaiToken` will disable it. Do you see it in different way?

Reply:
No, this approach should work fine.